classes/class3/homework_readable/app.js

```diff
@@ -0,0 +1,44 @@
+var express = require('express');
+var path = require('path');
+var index = require('./routes/index');
+var logger = require('morgan');
+var cookieParser = require('cookie-parser');
+var bodyParser = require('body-parser');
+var exphbs = require('express-handlebars');
+var mongoose = require('mongoose');
+var mongo = require('mongodb');
+
+var mongoURI = process.env.MONGOURI || "mongodb://localhost/test";
+mongoose.connect(mongoURI);
+
+var PORT = process.env.PORT || 3000;
+
+var app = express();
+
+
+app.engine('handlebars', exphbs({defaultLayout: 'main'}));
+app.set('view engine', 'handlebars');
+
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
+
+app.get('/', function(req,res){
+	res.send('This is a cat app! What are you doing here?');
+});
+
+app.get('/cats', index.list_cats);
+
+app.get('/cats/new', index.make_cat);
+
+app.get('/cats/delete/old', index.remove_cat);
+
+app.get('/cats/bycolor/*', function(req,res){
```

Comment:
Instead of a `*` wild card, we can use `/cats/bycolor/:color` which will give us a conveniently named `req.params.color` object. It means we don't have to do any of the hacky stuff on the line below to get that parameter in to our route handler. Express has some nice conveniences like that!